include/range/v3/view/trim.hpp

```diff
@@ -118,24 +118,19 @@ namespace ranges
                 return {all(static_cast<Rng &&>(rng)),
                         compose(std::move(pred), std::move(proj))};
             }
-        };
-
-        struct trim_fn : trim_base_fn
-        {
-            using trim_base_fn::operator();
 
             template<typename Pred>
             constexpr auto operator()(Pred pred) const
             {
-                return make_view_closure(bind_back(trim_base_fn{}, std::move(pred)));
+                return make_view_closure(bind_back(trim_fn{}, std::move(pred)));
             }
             template<typename Pred, typename Proj>
             constexpr auto CPP_fun(operator())(Pred && pred,
                                                Proj proj)(const //
                                                           requires(!range<Pred>))
             {
                 return make_view_closure(bind_back(
-                    trim_base_fn{}, static_cast<Pred &&>(pred), std::move(proj)));
+                    trim_fn{}, static_cast<Pred &&>(pred), std::move(proj)));
```

Comment:
The idea of using `trim_base_fn` here (and similar pattern elsewhere) was to narrow the number of `operator()` overloads that must be considered once partial application has happened. I would like to preserve this pattern.

Reply:
Yes, I got that. This was an expedient fix. For the peanut gallery: I'll be reworking this PR to compose overload sets from multiple bases to avoid the [namespace.udecl] problem as @ericniebler and I discussed offline.